src/main/java/org/jenkinsci/plugins/github/config/GitHubPluginConfig.java

```diff
@@ -270,13 +284,29 @@ private static void validateConfig(boolean state, String message) {
         }
     }
 
+    @Deprecated
     public HookSecretConfig getHookSecretConfig() {
-        return hookSecretConfig;
+        return hookSecretConfigs != null && !hookSecretConfigs.isEmpty()
+                ? hookSecretConfigs.get(0)
+                : new HookSecretConfig(null);
     }
 
-    @DataBoundSetter
```

Comment:
will it break existing caas configs?

Reply:
Not sure specifically how CasC works here. In general it does not guarantee backward compatibility even when settings XML is compatible. Out of scope for the design.

Reply:
it would be bad to break casc users, such change would require some version and compat hpi marker...

Reply:
`compatibleSinceVersion` shows up in the GUI, not useful for CasC users.

In general when updating core/plugins with CasC you may need to make adjustments to YAML to match. AFAIK you get a straightforward error message otherwise so it is pretty obvious.